include/revng/Pypeline/ObjectIDImpl.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <set>
+
+#include "boost/functional/hash.hpp"
```

Comment:
We are trying to dismiss `boost`. No new includes unless there's a *strong* reason.
Use `hashCombine`.